spec/requests/addresses_controller_spec.rb

```diff
@@ -0,0 +1,123 @@
+require 'rails_helper'
+
+RSpec.describe "AddressesControllers", type: :request do
+  let!(:user) { FactoryBot.create(:user) }
+  let!(:address) { FactoryBot.create(:address, user: user) }
+  
+  before do
+    allow(Current).to receive(:user).and_return(user)
+  end
+
+  describe "GET /new" do
+    it "renders the new template of address" do
+      get new_user_address_path(user.id)
+      expect(response).to render_template('new')
+    end  
+  end
+
+  describe "POST /create" do
+    it "creates a new address" do
+      params = {
+        address: {
+          area: '268G, Sneh Nagar',
+          city:  'Jabalpur',
+          pincode:  '480001', 
+          state: 'MP'
+        }
+      }
+      post user_addresses_path(user.id), params: params
```

Comment:
We should also check something like this, `expect { post user_addresses_path(user.id), params: params }.to change { Address.count }.by(1)`

To check that this adds an address to the database.